lib/model/folder_sendrecv.go

```diff
@@ -1109,7 +1156,14 @@ func (f *sendReceiveFolder) copierRoutine(in <-chan copyBlocksState, pullChan ch
 			l.Debugf("not weak hashing %s. not enough changed %.02f < %d", state.file.Name, blocksPercentChanged, f.WeakHashThresholdPct)
 		}
 
+	blocks:
 		for _, block := range state.blocks {
+			select {
+			case <-f.ctx.Done():
+				break blocks
+			default:
+			}
+
```

Comment:
Isn't this breakout imply we're done with success or something along those lines?

Reply:
Should we state.fail(f.ctx.Err()) ?

Reply:
It shouldn't be necessary because the finisher only closes and "commits" a file if all blocks are accounted for, which isn't the case if we abort early here. However I agree, it's probably a good idea to make it explicit.

Reply:
I guess you might get "Failed items" which is not ideal, but I am somehow afraid this will under some circumstances assume we're done downloading and rename the file to place.

Reply:
The folder won't disappear from the UI, nor will the events about failed items in the buffers.

Reply:
Failed items are queried from a dedicated rest endpoint and are stored in `sendReceiveFolder.errors`, so they do get reset. Even if pulling on folder stop takes a long time, the folder error events would only be issued at the end of one puller iteration, but it will abort before that when the context is cancelled.